oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -215,9 +215,9 @@ protected Mappings createMapping(Model model) {
                 properties.put(matchCName, matchColumn);
             } else {
                 Map<String, Object> column = new HashMap<>();
-                column.put(
-                    "type", columnTypeEsMapping.transform(columnDefine.getType(), columnDefine.getGenericType()));
-                if (columnDefine.isStorageOnly()) {
+                column.put("type", type);
+                // no index parameter is allowed for binary type, since ES 8.0
```

Comment:
Curiously, What binary fields are we setting `parameter`? And which parameters did we set?

Reply:
Parameter name is "index", examples are "tags"

Reply:
https://github.com/apache/skywalking/blob/f7bcb76784f968b865f31c7abfbd7c3a33945986/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/alarm/AlarmRecord.java#L80

Reply:
Oh, I see. It actually should add `storageOnly = true`, this is also a code bug.

Reply:
> Oh, I see. It actually should add `storageOnly = true`, this is also a code bug.

I can fix it, but FYI, regardless of that even we set `index=false`, it's still not allowed in ES8 to have parameter `index` in binary type.

Reply:
> Oh, I see. It actually should add `storageOnly = true`, this is also a code bug.

Fixed in this patch.